tests/testthat/test-slice.R

```diff
@@ -446,10 +446,53 @@ test_that("slicing an unspecified() object returns an unspecified()", {
 
 
 test_that("vec_slice() works with Altrep classes with custom extract methods", {
-  x <- .Call(vctrs_altrep_rle_Make, c(foo = 10L, bar = 5L))
+  x <- chr_rle(foo = 10L, bar = 5L)
```

Comment:
Note this only tests the barrier altrep path (character / list). We don't have an altrep test class for the non-barrier path (like an altrep integer rep or something).

We could make one if we feel like we need the coverage